### PR TITLE
Add microarray gene ID conversion example

### LINKED ISSUE #113: Create microarray gene ID conversion example
This is a parallel issue to #110 

The only module I'm seeing that has an RNA-seq example and not a microarray example is the gene ID conversion: ensembl-id-convert

I'm not sure that we will need to change it that much from the RNA-seq example except that it will need to be a different dataset.

### PR DESCRIPTION
**Pull request by cbethell**
## Purpose:
What issue(s) does your PR address?

This PR closes #113  (the issue on creating a microarray gene ID conversion example notebook).

## Strategy
What was your strategy for this new or edited analysis?

I followed the comments on issue #113 and found a new (larger than n = 4 samples) mouse microarray dataset.

The dataset used in this paper is a [mouse glioma cancer stem cell dataset](Cancer Stem Cells Are Enriched In The Side-Population Cells In A Mouse Model Of Glioma) with n = 15 samples (which has been uploaded to the S3 bucket for testing/review).

I then copied the analysis steps from the RNA-seq gene ID conversion notebook, but changed the annotation package being loaded in to that relevant to the mouse genome.

## Concerns/Questions for reviewers:
What things should reviewers look out for?

- Is the current workflow missing any vital steps?
- Do the steps under the `Explore mapped data frame` section seem helpful? Are the added steps clear? Is there anything that should be added/removed from this section?

### Content checks
* [x] All `{{BLANKS}}` have been replaced with the correct content.
* [x] [Sources are cited](https://github.com/AlexsLemonade/refinebio-examples/blob/master/CONTRIBUTING.md#citing-sources-in-text)
* [x] Seed is set (if applicable)

### Formatting Checks
* [x] Removed any manual numbering of sections.
* [x] Removed any instances of chunk naming.
* [x] Spell checked any Rmd file or md file.
* [x] Comments and documentation are up to date.

### Add datasets to S3
* [x] Added [data and metadata files to S3](https://github.com/AlexsLemonade/refinebio-examples/blob/master/CONTRIBUTING.md#adding-datasets-to-the-S3-bucket).

### Docker/Snakemake rendering components
* [x] Added the `.html` [link to the navigation bar](https://github.com/AlexsLemonade/refinebio-examples/blob/master/CONTRIBUTING.md#add-new-analyses-to-the-navbar).
* [x] Any not yet added packages needed for this analysis have been added to the Dockerfile and it successfully builds.
* [x] In the Docker container, [snakemake was run for rendering](https://github.com/AlexsLemonade/refinebio-examples/blob/master/CONTRIBUTING.md#how-to-re-render-the-notebooks).

### CODE REVIEW

File: 02-microarray/gene-id-convert_microarray_01_ensembl.Rmd
Comment (cansavvy) [inline 02-microarray/gene-id-convert_microarray_01_ensembl.Rmd:248]: I briefly mentioned this on the issue, but I think toward the goal of modeling "always look at your data" it may be good for us to include a few quick investigation steps that can tell a bit about "how the mapping went". This can include but isn't limited to answering: 
- How many gene identifiers are mapped versus not mapped?
- How many multi mappings does each gene identifier have?

File: 02-microarray/gene-id-convert_microarray_01_ensembl.Rmd
Comment (cansavvy) [inline 02-microarray/gene-id-convert_microarray_01_ensembl.Rmd:233]: I think we will need to hit this point harder and briefly mention the different options for `multiVals` argument, highlighting that different downstream analyses may benefit from particular strategies. Whatever strategy we end up using here, we should also use a different one when we update the rna-seq conversion module.

**Review by cansavvy — COMMENTED** [review]
This is a good start!  This doesn't yet discuss the `multivals` argument that I mentioned here: https://github.com/AlexsLemonade/refinebio-examples/pull/212#discussion_r486332840 so the last check in you did isn't exactly reflective of what's going on with the mappings. 

I think the meaning of the `multivals` argument and how it influences the output is a worthwhile thing for us to dive into. Because if you are confused by this argument, our users will definitely be confused by this too (I don't like that they have a default for this, I think it makes it easy for people to miss it). We should aim for making the `multivals` argument and what it means, _as clear as possible_ for our users.

So I think next you may want to take another closer look at the `mapIds` docs and we can always schedule a zoom/Google hangout/Slack/Teams chat to discuss if that would be helpful.

**Comment by cbethell** [comment]
> This is a good start! This doesn't yet discuss the `multivals` argument that I mentioned here: [#212 (comment)](https://github.com/AlexsLemonade/refinebio-examples/pull/212#discussion_r486332840) so the last check in you did isn't exactly reflective of what's going on with the mappings.
> 
> I think the meaning of the `multivals` argument and how it influences the output is a worthwhile thing for us to dive into. Because if you are confused by this argument, our users will definitely be confused by this too (I don't like that they have a default for this, I think it makes it easy for people to miss it). We should aim for making the `multivals` argument and what it means, _as clear as possible_ for our users.
> 
> So I think next you may want to take another closer look at the `mapIds` docs and we can always schedule a zoom/Google hangout/Slack/Teams chat to discuss if that would be helpful.

Per a discussion with @cansavvy re the above comment, the plan moving forward is as follows:
- Supply the `multiVals` argument with the `"list"` option
- Use `reshape2::melt()` to get that large list into a more manageable data frame
- Explore the melted data frame using some of the steps already used in this PR (to look at multi mappings, etc.)
- Then rerun the `mapIds()` function, this time supplying the `multiVals` argument with the`"filter"` option which will remove all elements with multiple matches

**Comment by cbethell** [comment]
@cansavvy, I _believe_ that I implemented the plan that we discussed and outlined in the [last comment of this PR](https://github.com/AlexsLemonade/refinebio-examples/pull/212#issuecomment-692251653).

I also implemented the suggestions you provided through your most recent review of this PR (the main ones being changing the file name, keeping the use of `IDs` consistent throughout the notebook, and some sentence restructuring).

Please let me know if I misinterpreted any of your suggestions or missed any!

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cansavvy) [inline 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:144]:
```suggestion
Although this example notebook uses Ensembl IDs from Mouse, (<i>Mus musculus</i>), to obtain gene symbols this notebook can be easily converted for use with different species or annotation types <i>eg</i> protein IDs, gene ontology, accession numbers.
```

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cansavvy) [inline 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:147]: I know its' not ready yet, but we should probably have this go to the html instead of the Rmd.

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cansavvy) [inline 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:164]:
```suggestion
In this analysis, we will be using the `org.Mm.eg.db` R package [@Carlson2019].
[Other species can be used](#using-a-different-refinebio-dataset-with-this-analysis). 
```

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cansavvy) [inline 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:223]: This message will change depending on the `multiVals` argument, so we either need to reframe this sentence or delete it. If we do keep it, we should put it after the user would see this message.

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cansavvy) [inline 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:224]: I like this section! Great!

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cansavvy) [inline 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:253]: Nice! I like this too!

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cansavvy) [inline 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:258]: I *think* you can use the `value` arugment to name what you'd like the column to be called, but you'd have to try that. I think we should either do that or use `rename()` so its clear what these columns are.

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cansavvy) [inline 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:286]: Yeah, we might want to rename `L1` before we get to this step.

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cansavvy) [inline 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:300]:
```suggestion
If we are not too worried about the 130 IDs with multiple mappings, we can filter them out for the purpose of having 1:1 mappings for our downstream analysis.
```

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cansavvy) [inline 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:304]:
```suggestion
The next code chunk we will rerun the `mapIds()` function, this time supplying the `"filter"` option to the `multiVals` argument.
```

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cansavvy) [inline 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:320]: Yes, this bit, can we do this above?

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cansavvy) [inline 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:318]: I might be wrong, but I don't think this melt is necessary, I *think* you should be able to put this straight into a data frame since it should be the same length as the original data.

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cansavvy) [inline 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:323]: I don't like copying the code. I also don't think this step is necessary. We can trust `mapIds()` will work as advertised. Plus this notebook has gotten a tad long. I would go straight to writing to TSV and call it a day.

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cansavvy) [inline 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:357]: I wouldn't share this alternative annotation package's user guide, I find it more confusing than helpful. Easier just to share AnnotationDbi user guide tutorials as opposed to sending our users to something else that they would have to learn from scratch.

**Review by cansavvy — COMMENTED** [review]
This looks great! Just have a few comments about a couple things to rearrange, I think this is close!

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cbethell) [reply 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd]: Gotcha, that makes sense. I removed this reference but haven't found any additional user guide tutorials thus far, do you have any suggestions or personal favorites for this particular case?

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cansavvy) [inline 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:305]:
```suggestion
The next code chunk we will rerun the `mapIds()` function, this time supplying the `"filter"` option to the `multiVals` argument.
```

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cansavvy) [inline 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:311]: Should probably save `data` with it too.

Comment (cansavvy) [inline 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:311]:
```suggestion
filtered_mapped_df <- data.frame(
     df, 
```
This isn't formatted properly, but you get the idea.

**Review by cansavvy — APPROVED** [review]
Two tiny-ish comments! LGTM after that!

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cbethell) [reply 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:311]: @cansavvy, I get an error here due to the difference in length between `df` (nrows = 17918) and the results in `filtered_mapped_df` (nrows = 17854). Do you think we should pull the genes out of the rownames and do an `inner_join` here?

Comment (cansavvy) [reply 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:311]: Oh of course. Forgot about that. Will need an inner_join. I can take a look after add that if you want.

Comment (cbethell) [reply 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:311]: Okay @cansavvy, you can take a look at the `inner_join` step and let me know if I need to make any changes 👍

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cansavvy) [inline 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:244]:
```suggestion
Now, let's take a look at our `mapped_list` to see how the mapping went.
```

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cansavvy) [inline 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:317]:
```suggestion
    multiVals = "filter" # This will drop any `Gene`s that have multiple matches
```

**Review by cansavvy — APPROVED** [review]
Just a couple little comments. Looks like there's some template changes that might need to be done too; some that got lost: https://github.com/AlexsLemonade/refinebio-examples/pull/216#discussion_r490194300
But I will probably file a separate PR to change these everywhere since this will affect every module and I haven't yet figured out the full extent of what needs to be changed.

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cansavvy) [inline 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd:14]: Probably should have a citation here too.

File: 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd
Comment (cansavvy) [reply 02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd]: I don't think I have anything else. We can add more if we find something.

**Review by jaclyn-taroni — APPROVED** [review]
LGTM

**Comment by jaclyn-taroni** [comment]
I'm playing with the settings that require branches to be up to date before merging. I am going to merge this despite the lack of status check (that I do not currently understand).

**Pull request merged**
